terraform/node_provider.go

```diff
@@ -1,11 +1,114 @@
 package terraform
 
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/terraform/configs/configschema"
+	"github.com/hashicorp/terraform/providers"
+	"github.com/hashicorp/terraform/tfdiags"
+)
+
 // NodeApplyableProvider represents a provider during an apply.
 type NodeApplyableProvider struct {
 	*NodeAbstractProvider
 }
 
-// GraphNodeEvalable
-func (n *NodeApplyableProvider) EvalTree() EvalNode {
-	return ProviderEvalTree(n, n.ProviderConfig())
+var (
+	_ GraphNodeExecutable = (*NodeApplyableProvider)(nil)
+)
+
+// GraphNodeExecutable
+func (n *NodeApplyableProvider) Execute(ctx EvalContext, op walkOperation) error {
+	_, err := ctx.InitProvider(n.Addr)
+	if err != nil {
+		return err
+	}
+	provider, _, err := GetProvider(ctx, n.Addr)
+	if err != nil {
+		return err
+	}
+
+	switch op {
+	case walkValidate:
+		//Validate
+		return n.ValidateProvider(ctx, provider)
+	case walkRefresh, walkPlan, walkApply, walkDestroy:
+		// EvalConfigProvider
```

Comment:
This comment (EvalConfigProvider) is helpful for this review, but I think is confusing now as it refers to code that's now deleted